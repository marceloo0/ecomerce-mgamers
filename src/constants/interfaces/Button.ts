interface Common {
  size?: 'small' | 'medium' | 'large'
}

export interface ButtonProps extends Common {
  icon?: JSX.Element
  children?: React.ReactNode
  onClick?: () => void
}

export interface WrapperProps extends Common {
  hasIcon: boolean
  hasChildren: boolean
}
