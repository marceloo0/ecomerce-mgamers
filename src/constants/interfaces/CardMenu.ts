export interface CardMenuImgStyle {
  sizeImg: 'small' | 'medium' | 'large'
}

export interface CardMenuProps {
  img: string
  price: string
  titleBackground: string
  onClick?: () => void
}
