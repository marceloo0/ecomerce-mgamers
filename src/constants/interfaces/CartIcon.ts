export interface CartIconProps {
  icon: React.ReactNode
}
