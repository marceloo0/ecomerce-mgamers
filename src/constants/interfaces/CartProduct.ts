export interface CartProductProps {
  price: string
  promotion: string
  title: string
  img: string
  titleBackground: string
  handleNextSlide: () => void
  handlePrevSlide: () => void
}

export interface CartItemProps extends CartProductProps {
  quantity: number
}

const propsCardTest = {
  price: '109',
  titleBackground: 'teclados',
  img: '/img/mouseMenu.svg'
}

export const props = {
  price: 'R$ 109,00',
  titleBackground: 'teclados',
  img: '/img/products/mouse.svg',
  promotion: 'R$ 89,00',
  title: ''
}
