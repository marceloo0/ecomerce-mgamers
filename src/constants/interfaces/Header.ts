export interface HeaderProps {
  setShowMenu?: () => void
  login?: boolean
}
