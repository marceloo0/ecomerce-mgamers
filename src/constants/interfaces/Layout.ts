export interface Layout {
  children: React.ReactNode
}
