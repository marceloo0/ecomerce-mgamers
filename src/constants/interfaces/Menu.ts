export const menu = [
  {
    title: 'NovidadeS',
    img: '/img/menu/news.svg',
    price: 'R$ 100,00'
  },
  {
    title: 'CadeiraS',
    img: '/img/menu/fantech.svg',
    price: 'R$ 919,90'
  },
  {
    title: 'HeadseatS',
    img: '/img/menu/headset.svg',
    price: 'R$ 319,90'
  },
  {
    title: 'MouseS',
    img: '/img/menu/mouse.svg',
    price: 'R$ 79,90'
  },
  {
    title: 'MousepadS',
    img: '/img/menu/mousepad.svg',
    price: 'R$ 69,90'
  },
  {
    title: 'TecladoS',
    img: '/img/menu/teclado.svg',
    price: 'R$ 219,33'
  }
]
