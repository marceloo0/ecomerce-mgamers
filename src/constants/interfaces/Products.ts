export interface ProductsProps {
  id: number
  img: string
  price: number
  promotion: number
  title: string
  titleBackground: string
}
export interface Stock extends ProductsProps {
  quantity: number
}
export interface ButtonProductProps extends Stock {
  handleNextSlide: () => void
  handlePrevSlide: () => void
}
export interface ProductsItemProps extends ButtonProductProps {
  item: Stock
}
export interface ItemProductProps extends Stock {
  item: Stock
}
