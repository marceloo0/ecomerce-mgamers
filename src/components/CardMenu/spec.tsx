import { screen } from '@testing-library/react'
import { renderWithTheme } from '../../utils/tests/helpers'
import { props } from '../../constants/interfaces/CartProduct'

import { CardMenu } from '.'

describe('<CardMenu />', () => {
  it('should render default', () => {
    const { container } = renderWithTheme(<CardMenu {...props} />)

    expect(screen.getByText('R$ 109,00')).toBeInTheDocument()

    expect(screen.getByText(/teclados/i)).toBeInTheDocument()

    expect(
      screen.getByRole('img', { name: props.titleBackground })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render titleBackground', () => {
    renderWithTheme(
      <CardMenu
        price="R$ 109,00"
        titleBackground="teclados"
        img="/img/mouseMenu.svg"
      />
    )

    expect(screen.getByText(/teclados/i)).toBeInTheDocument()
  })
})
