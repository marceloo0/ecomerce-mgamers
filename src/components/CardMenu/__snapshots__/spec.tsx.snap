// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardMenu /> should render default 1`] = `
.c0 {
  border: 1px solid #7FFA45;
  height: 350px;
  position: relative;
  overflow: hidden;
  background-color: #201C2B;
  border-radius: 0.4rem;
  cursor: pointer;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c0:hover {
  box-shadow: rgb(0 0 0 / 80%) 0px 40px 58px -16px,rgb(0 0 0 / 72%) 0px 30px 22px -10px;
}

.c1 {
  position: absolute;
  left: 1rem;
  top: 1rem;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  color: #C7ECEC;
  font-size: 1.2rem;
}

.c3 {
  font-weight: bold;
  color: #25AF43;
  font-size: 3.8rem;
}

.c4 {
  height: 360px;
  width: 380px;
  position: absolute;
  right: -100px;
  top: -50px;
  z-index: 1;
  object-fit: contain;
}

.c5 {
  position: absolute;
  left: 0;
  bottom: 0;
  z-index: 0;
  font-size: 5.8rem;
  color: #2E2F42;
  font-weight: bold;
}

<button
  class="c0"
>
  <div
    class="c1"
  >
    <span
      class="c2"
    >
      Apartir de
    </span>
    <span
      class="c3"
    >
      R$ 109,00
    </span>
  </div>
  <img
    alt="teclados"
    class="c4"
    src="/img/products/mouse.svg"
  />
  <span
    class="c5"
  >
    teclados
  </span>
</button>
`;
